### ISSUE TITLE
Valid docs rule does not support closure parameters

### ISSUE THREAD

**Issue by diogoguimaraes** [realm/SwiftLint#267]
The rule that validates documentation blocks does not support closure type parameters and mistakes them with return values. 
I think it is probably related to this func:
```
func delcarationReturns(declaration: String, kind: SwiftDeclarationKind) -> Bool {
    return declaration.containsString("->") || SwiftDeclarationKind.variableKinds().contains(kind)
}
```
The `->` that exists in the closure parameter is mistaken by the return `->`.

**Comment by kevindelord** [comment]
+1

I just updated Swiftlint and no function with a closure parameter can be properly documented without a warning.

**Comment by jpsim** [comment]
Sorry for the false positive here. This is a bug and should be fixed. Help doing so is appreciated!

**Comment by proth** [comment]
:+1:

**Comment by ecnepsnai** [comment]
I'm still seeing this issue on 0.9.2

```swift
    /// Performs a simple GET request to the specified API path
    ///
    /// - parameters:
    ///   - path: The API path
    ///   - finished: Called when a response (failure of successful) has been received
    func APIrequest(path: String, finished: (NSError?, JSON) -> (Void)) {
```

It fails the valid_docs validation, and can be solved by adding a ` - returns: foo` line to the doc even though this function is not returning anything.
